instance/factories.py

```diff
@@ -168,6 +168,25 @@ def production_instance_factory(**kwargs):
     )
     instance_kwargs.update(kwargs)
 
+    assert is_unique_domain(instance_kwargs)
+
     # Create instance
     production_instance = OpenEdXInstance.objects.create(**instance_kwargs)
     return production_instance
+
+
+def is_unique_domain(instance_kwargs):
+    """
+    Uses the same logic as the constructor for the `DomainNameInstance`
+    mixin to determine what the instance's internal_lms_domain will be
+    once created, and then performs a case-insensitive check to see if
+    any other instances have the same domain name.
+    """
+    if 'internal_lms_domain' in instance_kwargs:
+        internal_lms_domain = instance_kwargs['internal_lms_domain']
+    else:
+        sub_domain = instance_kwargs['sub_domain']
+        internal_lms_domain = generate_internal_lms_domain(sub_domain)
+
+    count = OpenEdXInstance.objects.filter(internal_lms_domain__iexact=internal_lms_domain).count()
```

Comment:
You can just use:
```
    OpenEdXInstance.objects.filter(internal_lms_domain__iexact=internal_lms_domain).exists()
```

Reply:
Removed in #8be0f2c94009d88c2ba1cf4adacbce7c335a73db